src/Configuration/Config.KeyPerFile/test/KeyPerFileTests.cs

```diff
@@ -217,6 +217,61 @@ void ReloadLoop()
             Assert.Equal("Foo", options.Text);
         }
 
+        [Fact]
```

Comment:
Can you add a test where nothing exists to start, and the directory/files get created later so its the nothing found to something case as opposed to just modification scenarios?

Reply:
@Haok I added the test.